objectstore/objectstore.go

```diff
@@ -0,0 +1,58 @@
+package objectstore
+
+import (
+	"io"
+
+	"github.com/fastly/compute-sdk-go/internal/abi/fastly"
+)
+
+type Entry struct {
+	io.Reader
+
+	validString bool
+	s           string
+}
+
+func (e *Entry) String() string {
+	if e.validString {
+		return e.s
+	}
+
+	// TODO(dgryski): replace with StringBuilder + io.Copy ?
+	b, err := io.ReadAll(e)
+	if err != nil {
+		return ""
+	}
+
+	e.validString = true
+
+	return string(b)
+
+}
+
+// Store represents a Fastly object store
+type Store struct {
+	objectstore *fastly.ObjectStore
+}
+
+// Open returns a handle to the named object store
+func Open(name string) (*Store, error) {
+	o, err := fastly.OpenObjectStore(name)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Store{objectstore: o}, nil
+}
+
+// Lookup fetches a key from the associated object store.
+func (s *Store) Lookup(key string) (*Entry, error) {
+	val, err := s.objectstore.Lookup(key)
```

Comment:
Non-Blocking Comment:
We could provide more specific errors here if we wanted to:

- If the key is empty
- If the key is more than 1024 characters
- If the key contains one or more of these characters`\n\r[]*?#`
- If the key starts with `.well-known/acme-challenge/`
- If the key is exactly the string `.` or the string `..`